app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/miband/BatteryInfo.java

```diff
@@ -15,20 +23,45 @@ public int getLevelInPercent() {
         return 50; // actually unknown
     }
 
-    public String getStatus() {
+    public BatteryState getState() {
         if (mData.length >= 10) {
             int value = mData[9];
             switch (value) {
-                case 1:
-                    return GBApplication.getContext().getString(R.string.battery_low);
-                case 2:
-                    return GBApplication.getContext().getString(R.string.battery_medium);
-                case 3:
-                    return GBApplication.getContext().getString(R.string.battery_full);
-                case 4:
-                    return GBApplication.getContext().getString(R.string.battery_not_charging);
+                case DEVICE_BATTERY_NORMAL:
+                    return BatteryState.BATTERY_NORMAL;
+                case DEVICE_BATTERY_LOW:
+                    return BatteryState.BATTERY_LOW;
+                case DEVICE_BATTERY_CHARGING:
+                    return BatteryState.BATTERY_CHARGING;
+                case DEVICE_BATTERY_CHARGING_FULL:
+                    return BatteryState.BATTERY_CHARGING_FULL;
+                case DEVICE_BATTERY_CHARGE_OFF:
+                    return BatteryState.BATTERY_NOT_CHARGING_FULL;
             }
         }
-        return GBApplication.getContext().getString(R.string._unknown_);
+        return BatteryState.UNKNOWN;
+    }
+
+    public GregorianCalendar getLastChargeTime() {
```

Comment:
Do we have a central place where such dates are parsed? I think we have at least one other place where we do this. Can you centralize this (perhaps MiBandConst)?